### PR TITLE
Allow lambda@edge in aws-lambda-function module

### PR DESCRIPTION
**Pull request by palasha**
I want to use this module to create lambdas for cloudfront. That
requires supporting lambda@Edge. The two main changes here are:
1. Adding edgelambda as an authorized service for the role
2. granting permission create/update permission on * log groups.

The second is I suppose slightly controversial. because its an edge
lambda, we would have to precreate the log groups in every single aws
region, and I have no clue how to do that. I think this is an okay
tradeoff for now, and tried to document that in the description.

Test Plan:
wait for CI
use it in stacked TF and see if it works.

### CODE REVIEW

**Comment by palasha** [comment]
make fmt to fix lint

**Comment by palasha** [comment]
boolean -> bool

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/44dfb03f1a68c37245ae7a5ce9d81e5bbdfc0819..d4d6e0ebf1a3667d7b7af74239b2369d941ea4b6).

File: aws-cloudfront-naked-domain-redirect/variables.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
+  type        = string
+  description = "Env for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable service {
```
Comment (cziie) [inline aws-cloudfront-naked-domain-redirect/variables.tf:11]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "service" is declared but not used

File: aws-cloudfront-naked-domain-redirect/variables.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Service for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable owner {
```
Comment (cziie) [inline aws-cloudfront-naked-domain-redirect/variables.tf:16]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "owner" is declared but not used

File: aws-cloudfront-naked-domain-redirect/variables.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+variable project {
```
Comment (cziie) [inline aws-cloudfront-naked-domain-redirect/variables.tf:1]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "project" is declared but not used

File: aws-cloudfront-naked-domain-redirect/variables.tf [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
```
Comment (cziie) [inline aws-cloudfront-naked-domain-redirect/variables.tf:6]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "env" is declared but not used

File: aws-lambda-edge-add-hsts-header/main.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
+data "archive_file" "lambda" {
```
Comment (cziie) [inline aws-lambda-edge-add-hsts-header/main.tf:1]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Missing version constraint for provider "archive" in "required_providers"

Comment (cziie) [inline aws-lambda-edge-add-hsts-header/main.tf:1]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
data "archive_file" "lambda" is declared but not used

File: aws-lambda-edge-add-hsts-header/variables.tf [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
+  type        = string
+  description = "Env for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable service {
```
Comment (cziie) [inline aws-lambda-edge-add-hsts-header/variables.tf:11]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "service" is declared but not used

File: aws-lambda-edge-add-hsts-header/variables.tf [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Service for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable owner {
```
Comment (cziie) [inline aws-lambda-edge-add-hsts-header/variables.tf:16]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "owner" is declared but not used

File: aws-lambda-edge-add-hsts-header/variables.tf [hunk 8]
```diff
@@ -0,0 +1,19 @@
+variable project {
```
Comment (cziie) [inline aws-lambda-edge-add-hsts-header/variables.tf:1]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "project" is declared but not used

File: aws-lambda-edge-add-hsts-header/variables.tf [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  type        = string
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
```
Comment (cziie) [inline aws-lambda-edge-add-hsts-header/variables.tf:6]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "env" is declared but not used

File: aws-lambda-function/outputs.tf [hunk 10]
```diff
@@ -2,6 +2,10 @@ output arn {
   value = aws_lambda_function.lambda.arn
 }
 
+output qualified_arn {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:5]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`qualified_arn` output has no description

File: aws-lambda-function/outputs.tf [hunk 11]
```diff
@@ -2,6 +2,10 @@ output arn {
...
+  value = aws_lambda_function.lambda.qualified_arn
+}
+
 output invoke_arn {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:9]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`invoke_arn` output has no description

**Comment by palasha** [comment]
add publish argument and fix lint

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/a199d9534425024fd05f374dc63775437531df51..ff72a9cbe71d29c24d9ca1123d5fe94774c65694).

**Comment by palasha** [comment]
update README

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/7f37decedab836deafb8f746121e06c302a8c121..6d93c590224136065fe39d399c6ccfd6ea9a0713).

**Comment by ryanking** [comment]
@palasha given that your test plan includes some manual testing, can you let me know when that is done?

**Comment by palasha** [comment]
> @palasha given that your test plan includes some manual testing, can you let me know when that is done?

Sorry for the delay here! The manual testing is that:
- this actively working in production for the comealong.org website to add security headers.
- I've verified that no cloudwatch logs are being created in different regions.

**Comment by palasha** [comment]
> @palasha given that your test plan includes some manual testing, can you let me know when that is done?

Sorry for the delay here! The manual testing is that:
- this actively working in production for the comealong.org website to add security headers.
- I've verified that no cloudwatch logs are being created in different regions.

**Review by ryanking — APPROVED** [review]

**Pull request merged**